simulator/data_gen.py

```diff
@@ -0,0 +1,332 @@
+"""Module that creates randomly generated data to sent to the simulator IMU"""
+
+import csv
+from pathlib import Path
+
+import numpy as np
+import numpy.typing as npt
+import yaml
+from scipy.spatial.transform import Rotation as R
+
+from airbrakes.data_handling.imu_data_packet import (
+    EstimatedDataPacket,
+    RawDataPacket,
+)
+from constants import ACCELERATION_NOISE_THRESHOLD, GRAVITY
+from utils import deadband
+
+
+class DataGenerator:
+    """
+    Uses config settings to generate realistic flight trajectories. Returns raw and
+    estimated data points for the sim IMU to combine into data packets
+    """
+
+    __slots__ = (
+        "_config",
+        "_current_timestamp",
+        "_last_est_packet",
+        "_last_raw_packet",
+        "_last_velocity",
+        "_thrust_data",
+        "_vertical_index",
+    )
+
+    def __init__(self):
```

Comment:
it would be nice if we could specify the frequency that sim_imu runs at/this generates data at. I don't think it would be that much work to implement and it will be a lot easier to do it now rather than later.

Reply:
not sure what you mean by this. Do you mean like, real-time simulation, vs going full speed? Or do you mean the frequency between raw and estimated data packets? If the latter, then I have constants for that in the yaml config, and those are being used to determine the frequency between data packets.